src/napari/_qt/layer_controls/widgets/_tracks/qt_hide_completed_tracks_checkbox.py

```diff
@@ -0,0 +1,59 @@
+from qtpy.QtWidgets import (
+    QCheckBox,
+    QWidget,
+)
+
+from napari._qt.layer_controls.widgets.qt_widget_controls_base import (
+    QtWidgetControlsBase,
+    QtWrappedLabel,
+)
+from napari.layers import Tracks
+from napari.utils.translations import trans
+
+
+class QtHideCompletedTracksCheckBoxControl(QtWidgetControlsBase):
+    """
+    Class that wraps the connection of events/signals between the hide completed
+    tracks attribute and Qt widgets.
+
+    Parameters
+    ----------
+    parent: qtpy.QtWidgets.QWidget
+        An instance of QWidget that will be used as widgets parent
+    layer : napari.layers.Tracks
+        An instance of a napari Tracks layer.
+
+    Attributes
+    ----------
+    hide_completed_tracks_checkbox : qtpy.QtWidgets.QCheckBox
+        Checkbox controlling if completed tracks should be hidden.
+    hide_completed_tracks_checkbox_label : napari._qt.layer_controls.widgets.qt_widget_controls_base.QtWrappedLabel
+        Label for showing the option checkbox.
+    """
+
+    def __init__(self, parent: QWidget, layer: Tracks) -> None:
+        super().__init__(parent, layer)
+        # Setup layer
+
+        # Setup widgets
+        self.hide_completed_tracks_checkbox = QCheckBox()
+        self.hide_completed_tracks_checkbox.stateChanged.connect(
+            self.change_hide_completed_tracks
+        )
+
+        self.hide_completed_tracks_checkbox_label = QtWrappedLabel(
+            trans._('hide completed:')
+        )
+
+    def change_hide_completed_tracks(self, state: int) -> None:
+        self._layer.hide_completed_tracks = (
+            self.hide_completed_tracks_checkbox.isChecked()
+        )
```

Comment:
With the above change, we can remove this. We're also missing the reverse connection:

```suggestion
    def _on_hide_completed_tracks_change(self) -> None:
        """Receive layer model hide_completed_tracks event and update the checkbox."""
        with qt_signals_blocked(self.hide_completed_tracks_checkbox):
            self.hide_completed_tracks_checkbox.setChecked(
                self._layer.hide_completed_tracks
            )
```

Reply:
Thanks for the review @brisvag ! I don't quite understand why you propose to remove `change_hide_completed_tracks` though... I commited your suggestion but re-added the function for now.

Reply:
The `connect_setattr` function I suggested to add above is doing this work of connecting the qt element to the layer (with all the boilerplate of ensuring event loops are avoided and converting qtchecked to a bool.

This was recently updated in https://github.com/napari/napari/pull/8244, you can see how it was used in a similar way for (for example) the `PreserveLabelsCheckBoxControl`.

So now we only need the connection *back* from layer to the qt element (which we might be able to treat similarly in the future...)